packages/data/src/world/mq/deku_tree_mq.yml

```diff
@@ -2,126 +2,282 @@
+    "MQ Deku Tree Room Before Compass Room": "true"
+"MQ Deku Tree Room Before Compass Room":
+  dungeon: DT
+  exits:
+    "MQ Deku Tree Room Before Compass Room Door to 2nd Floor": "true"
+    "MQ Deku Tree Compass Room Entry Door": "true"
+  events:
+    #STICKS: "can_kill_baba_sticks"  #Big Deku Baba here
+    #NUTS: "can_kill_baba_nuts" #Big Deku Baba here
+    DEKU_MQ_BEFORE_COMPASS_EYE_SWITCH: "can_use_slingshot || can_use_bow"
+  locations:
+    "MQ Deku Tree Heart Before Compass": "true"
+    "MQ Deku Tree Grass Room Before Compass 1": "can_cut_grass"
+    "MQ Deku Tree Grass Room Before Compass 2": "can_cut_grass"
+    "MQ Deku Tree Grass Room Before Compass 3": "can_cut_grass"
+    "MQ Deku Tree Grass Room Before Compass 4": "can_cut_grass"
+    "MQ Deku Tree Grass Room Before Compass 5": "can_cut_grass"
+    "MQ Deku Tree Grass Room Before Compass 6": "can_cut_grass"
+    "MQ Deku Tree Grass Room Before Compass 7": "can_cut_grass"
+"MQ Deku Tree Compass Room Entry Door":
+  dungeon: DT
+  exits:
+    "MQ Deku Tree Room Before Compass Room": "true"
+    "MQ Deku Tree Compass Room Exit Door": "event(DEKU_MQ_BEFORE_COMPASS_EYE_SWITCH)"
+"MQ Deku Tree Compass Room Exit Door":
+  dungeon: DT
+  exits:
+    "MQ Deku Tree Compass Room Entry Door": "true"
+    "MQ Deku Tree MQ Compass Room Entrance": "true"
+"MQ Deku Tree MQ Compass Room Entrance":
+  dungeon: DT
+  exits:
+    "MQ Deku Tree Compass Room Exit Door": "true"
+    "MQ Deku Tree MQ Compass Room Chest Side": "true"
+"MQ Deku Tree MQ Compass Room Chest Side":
   dungeon: DT
   exits:
-    "Deku Tree Lobby": "true"
-    "Deku Tree Compass Room Alcove": "has_bombchu || has_mask_blast || (can_play_time && has_explosives_or_hammer)"
+    "MQ Deku Tree MQ Compass Room Entrance": "can_use_slingshot || climb_anywhere || hookshot_anywhere"
+    "MQ Deku Tree MQ Compass Room Chest Side Alcove": "has_bombchu || has_mask_blast || (can_play_time && has_explosives_or_hammer) || hookshot_anywhere || climb_anywhere"
   locations:
     "MQ Deku Tree Compass Chest": "true"
     "MQ Deku Tree Grass Compass Room 1": "can_cut_grass"
     "MQ Deku Tree Grass Compass Room 2": "can_cut_grass"
     "MQ Deku Tree Grass Compass Room 3": "can_cut_grass"
     "MQ Deku Tree Grass Compass Room 4": "can_cut_grass"
-"Deku Tree Compass Room Alcove":
+"MQ Deku Tree MQ Compass Room Chest Side Alcove":
   dungeon: DT
   exits:
-    "Deku Tree Compass Room": "true"
+    "MQ Deku Tree MQ Compass Room Chest Side": "has_explosives_or_hammer || hookshot_anywhere || climb_anywhere"
   locations:
     "MQ Deku Tree GS Compass Room": "gs && can_collect_distance"
     "MQ Deku Tree Heart Compass Room": "true"
-"Deku Tree Water Room":
+"MQ Deku Tree 3rd Floor":
+  dungeon: DT
+  exits:
+    "MQ Deku Tree 2nd Floor": "true"
+    "MQ Deku Tree Lobby": "true"
+    "MQ Deku Tree Basement": "true"
+    "MQ Deku Tree Slingshot Room Entry Door": "has_fire || event(DEKU_MQ_MAIN_TORCH)"
```

Comment:
Considering it's just a floor switch you walk over to and press, this can just be true.

Reply:
it is "true" via the event itself being true. I don't want to have the whole thing as true if eventual actorizer or something affecting it.

Reply:
and it can just be changed then. the has_fire is completely redundant right now, and who knows if the switch rando will even still use the events since you'd need to dynamically change the requirements if it becomes a hold switch or hammer switch.

Reply:
But why delete it if the "true" is already technically there? Why delete it just to re-add it if we need it?